packages/cli/src/cliExecuteCall.ts

```diff
@@ -0,0 +1,92 @@
+import { getDefaultProvider, providers, Wallet } from 'ethers';
+import { z } from 'zod';
+import { assetChains } from '@/shared/enums';
+import {
+  numericString,
+  hexString,
+  chainflipAsset,
+  chainflipNetwork,
+} from '@/shared/parsers';
+import {
+  type ExecuteOptions,
+  executeCall,
+  type ExecuteCallParams,
+} from '@/shared/vault';
+import { askForPrivateKey, getEthNetwork } from './utils';
+
+export const schema = z
+  .intersection(
+    z.union([
+      z.object({
+        chainflipNetwork: z.literal('localnet'),
+        ethNetwork: z.string(),
+        srcTokenContractAddress: z.string(),
+        vaultContractAddress: z.string(),
+      }),
+      z.object({
+        chainflipNetwork,
+        ethNetwork: z
+          .string()
+          .optional()
+          .transform(() => undefined),
+        srcTokenContractAddress: z
+          .string()
+          .optional()
+          .transform(() => undefined),
+        vaultContractAddress: z
+          .string()
+          .optional()
+          .transform(() => undefined),
+      }),
+    ]),
+    z.object({
+      walletPrivateKey: z.string().optional(),
+      srcAsset: chainflipAsset.optional(),
+      destAsset: chainflipAsset,
+      amount: numericString,
+      destAddress: z.string(),
+      message: hexString,
+      gasAmount: numericString,
+    }),
+  )
+  .transform(({ destAsset, srcAsset, ...rest }) => ({
+    ...rest,
+    destChain: assetChains[destAsset],
+    destAsset,
+    ...(srcAsset && { srcAsset }),
+  }));
+
+export default async function cliExecuteCall(unvalidatedArgs: unknown) {
```

Comment:
so chainflip-backend calls this `executeSwapAndCall`. This does the same right? Why don't we use the same name?

Reply:
i used the naming of the contract because that is what is invoked: https://github.com/chainflip-io/chainflip-eth-contracts/blob/cd289f4b0f78a7a9ea55bc221e88bba528ee9ff4/contracts/Vault.sol#L347

Reply:
where did you find the `executeSwapAndCall`?